backend/src/case_file/case_file.service.ts

```diff
@@ -904,5 +909,193 @@ export class CaseFileService {
+          },
+        },
+      },
+    });
+
+    // construct the equipmentDetails list
+    actions.forEach((action) => {
+      const equipment = action.equipment;
+
+      // get the action xref for the action
+      let actionData = actionCodes.find(
+        (element) =>
+          element.action_type_action_xref_guid ===
+          action.action_type_action_xref_guid
+      );
+  
+      if (equipment) {
+        // Initialize or update the equipment detail entry in the map
+        let equipmentDetail =
+          equipmentDetailsMap.get(equipment.equipment_guid) ||
+          ({
+            equipmentGuid: equipment.equipment_guid,
+            actionEquipmentTypeCode: equipment.equipment_code,
+            actionEquipmentTypeActiveIndicator: equipment.active_ind,
+            address: equipment.equipment_location_desc,
+            xCoordinate: "",
+            yCoordinate: "",
+            actions: [],
+          } as Equipment);
+
+        // Append the action to this equipment's list of actions
+        equipmentDetail.actions.push({
+          actor: action.actor_guid,
+          date: action.action_date,
+          activeIndicator: action.active_ind,
+          actionCode: actionData.action_code,
+        });
+
+        equipmentDetailsMap.set(equipment.equipment_guid, equipmentDetail);
+      }
+    });
+    const equipmentDetails = Array.from(
+      equipmentDetailsMap.values()
+    ) as Equipment[];
+
+    return equipmentDetails;
+  }
+
+  async updateEquipment(
+    updateEquipmentInput: UpdateEquipmentInput
+  ): Promise<CaseFile> {
+
+    let caseFileOutput: CaseFile;
+    try {
+      await this.prisma.$transaction(async (db) => {
+      
+      });
+      } catch (exception) {
+        console.log("exception", exception);
+        throw new GraphQLError(
+          "Exception occurred. See server log for details",
+          {}
+        );
+      }
+
+    return caseFileOutput;
+  }
+
+  // create an equipment record - with actions to either set the equipment, or set and remove the equipment
+  async createEquipment(
+    createEquipmentInput: CreateEquipmentInput
+  ): Promise<CaseFile> {
+
+    let caseFileOutput: CaseFile;
+    try {
+      await this.prisma.$transaction(async (db) => {
+
+        let caseFile = await this.findOneByLeadId(
+          createEquipmentInput.leadIdentifier
+        );
+        let caseFileGuid;
+    
+        if (caseFile?.caseIdentifier) {
+          caseFileGuid = caseFile.caseIdentifier;
+        } else {
+          caseFileGuid = await this.createCase(createEquipmentInput);
+        }
+
+        const newEquipmentJSON = {
+          active_ind: true,
+          create_user_id: createEquipmentInput.createUserId,
+          create_utc_timestamp: new Date(),
+          update_user_id: createEquipmentInput.createUserId,
+          update_utc_timestamp: new Date(),
+          equipment_code:
+            createEquipmentInput.equipment[0].actionEquipmentTypeCode,
```

Comment:
are you 100% certain that the equipment array isn't going to be null? you might want to consider a guard to make sure there's an element in that array, or use the ?. operator when trying to get at those properties. a better idea would be to pull that first element off the array and use it directly instead of treating it as an array.